cmd/canary/canary-config/retro.go

```diff
@@ -0,0 +1,236 @@
+// Copyright (c) 2019, Waze, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+
+package canary_config
+
+import (
+	"errors"
+	"fmt"
+	"github.com/spf13/cobra"
+	"github.com/spinnaker/spin/cmd/gateclient"
+	"github.com/spinnaker/spin/util"
+	"net/http"
+	"strings"
+	"time"
+)
+
+type RetroOptions struct {
+	*canaryConfigOptions
+	output                  string
+	canaryConfigFile        string
+	controlGroup            string
+	controlLocation         string
+	experimentGroup         string
+	experimentLocation      string
+	startInstant            string
+	endInstant              string
+	extendedScopeParamsFile string
```

Comment:
What do you think about the idea of using [repeat flags](https://github.com/spf13/cobra/issues/661#issuecomment-377684634) for this instead of a file?

eg `spin retrocmd -ep '_scope_key=autoscaling-group-name' -ep '_location_key=region'`

then that that generates

`[]string{"_scope_key=autoscaling-group-name", "_location_key=region"}`

Which can be used to generate the JSON map for extended scope params.

Reply:
`pflag` has a nice `StringToStringVar` flag that lets you specify a map explicitly with `--my-flag="key1=value1,key2=value2,..."` so I went with that.